snap/snapenv/snapenv.go

```diff
@@ -63,6 +63,7 @@ func GetBasicSnapEnvVars(desc interface{}) []string {
 	return fillSnapEnvVars(desc, []string{
 		"SNAP={{.SnapPath}}",
 		"SNAP_DATA=/var{{.SnapPath}}",
+		"SNAP_SHARED_DATA=/var{{.SnapBasePath}}/shared",
```

Comment:
Note that touches a lot of tests as well. It may be better suited for another PR.